pkg/rsa/rsa.go

```diff
@@ -2,10 +2,176 @@ package rsa
 
 import (
 	"crypto"
+	"crypto/rand"
 	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
 	"fmt"
+	"io"
+	"os"
 )
 
+// sentinel error used for when PEM blocks can't be decoded
+var ErrPEMDecode = errors.New("error decoding PEM block")
+
+// rsa.GetRSAKey: reads or generates an RSA key pair at `privKeyPath` and `pubKeyPath` respectively, encrypted by `password`
+// returns: private key or error
+func GetRSAKey(privKeyPath string, pubKeyPath string, password string) (*rsa.PrivateKey, error) {
+	// try reading the private key
+	privkey, err := ReadPrivateKey(privKeyPath, password)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return nil, fmt.Errorf("rsa.GetKey: %w", err)
+		}
+		privkey, err = NewKeys(privKeyPath, pubKeyPath, password)
+		if err != nil {
+			return nil, fmt.Errorf("rsa.GetKey: %w", err)
+		}
+		return privkey, nil
+	}
+	// try reading the public key
+	pubkey, err := ReadPublicKey(pubKeyPath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return nil, fmt.Errorf("rsa.GetKey: %w", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("rsa.GetRSAKey: aes.ReadPublicKey: %w", err)
```

preferably the same for the rest of the errors. its not *needed* but it helps